content/blog/understanding-the-dom/index.md

```diff
@@ -0,0 +1,521 @@
+
+Let's see how this is done.
+
+At the root of any HTML file, you have three things: Tags, attributes, and text content.
+
+```html
+<!-- A "header" tag -->
+<header>
+  <!-- An "a" tag with an "href" attribute -->
+  <a href="example.com">
+    <!-- A text node -->
+    Example Site
+  </a>
+</header>
+```
+
+When you type a tag, like `header` or `a`, you're creating an _element node_. These nodes then compose to create _"leaves"_ on the DOM tree. Attributes are then able to manually add information to said nodes. When you have another element node inside of a seperate element node, you add a _"child"_ to said node. This relationship between the nodes allow you to preserve metadata between them, allows CSS to apply, and more.
+
+There's also the idea of a _"sibling"_ node. When a node's parent has more than one child, those other nodes are that child node's _"siblings"_.
+
+![A chart showing the relationships between parents and siblings](./dom_relationships.svg)
+
+Altogether, the terminologies used to refer between these node "leaves" are extremely similar to the terminology often used with family trees.
+
+There are some rules to this tree that's created from these "nodes":
+
+- There must be one "root" or "trunk" node, there cannot be more than one root
+- There must be a one-to-many relationship with parents and children
+	- A node may have many children
+	- But may not have many parents
+- A non-root node may have many siblings as a result of the parent having many children
+
+![A chart showing the aforementioned rules of the node relationships](./dom_relationship_rules.svg)
+
+### How It's Used By The Browser {#how-the-browser-uses-the-dom}
+
+This tree tells the browser all of the information the browser needs to execute tasks in order to display and handle interaction with the user. For example, when the following CSS is applied to an `index.html` file:
+
+```css
+// index.css
+#b li {
+	background: red;
+}
+```
+
+```html
+<!-- index.html -->
+<main id="a">
+	<p id="e"></p>
+  <ul id="b">
+    <li id="c"></li>
+    <li id="d"></li>
+  </ul>
+</main>
+```
+
+The browser is able to keep in mind, while it's moving through the tree, that it needs to find an element with the `ID` of `b` and then mark it's `li` children with a red background. They're "children" because the DOM tree keeps that relationship info that's defined by the HTML.
+
+![A chart showing the 'ul' tag highlighted in green with the children 'li' tags marked in red](./dom_tree_with_css.svg "Diagram showing the above code as a graph")
+
+> The `ul` element is marked as green just to showcase that it is the element being marked by the first part of the selector
+
+Typically, the browser will "visit" it's nodes in a specific order. For example, in the above chart, the browser might start at the `main` tag, then go to the `p` tag, then visit the `ul` tag, and finally the two children in order from left-to-right (`li id="c"` , `li id="d"`).
```

Comment:
```suggestion
Typically, the browser will "visit" it's nodes in a specific order. For example, in the above chart, the browser might start at the `<main>` tag, then go to the `<p>` tag, then visit the `<ul>` tag, and finally the two children in order from left-to-right (`<li id="c">` , `<li id="d">`).
```

Reply:
Even though the saying "`<tag>` tag" may sound like saying ATM machine, I believe it's still more clear, even this far into the document.

In case I miss some in my GitHub-side review, it may be nice to go back and double check them with
```
`\w+`
```